### PR TITLE
Feat/team invite link

### LINKED ISSUE #43: Create a url for inviting team members to your team
On the overview page, there should be an invitaiton link that only the team captain can see and can send out to their team mates. When the team mates click on this link, they should be redirected to the page where you create your account. When they have created their account, their account should be linked to the team so that when they sign in, they will see the correct team.

These links don't expire and can be reused as many times as you want. You can use a similar solution as with the Email Confirmation

### PR DESCRIPTION
**Pull request by Maxxen**
Closes #43 

Also includes misc fixes and tweaks to account creation / overview page / login views.

### CODE REVIEW

File: src/registrationSystem/views.py [hunk 1]
```diff
@@ -191,48 +193,89 @@ def activate(request, token):
...
+                is_utn_member=is_utn_member(form.cleaned_data["person_nr"]),
+            )
+
+            user.belongs_to_group = team
+            user.save()
+
+            return HttpResponseRedirect(reverse('overview'))
+        else:
+            print("ErrorS:")
+            print(form.errors.items())
```
Comment (dakaza98) [inline src/registrationSystem/views.py:271]: You can remove these prints

File: src/registrationSystem/views.py [hunk 2]
```diff
@@ -191,48 +193,89 @@ def activate(request, token):
...
 
-def temp(request):
-    return render(request, 'registrationSystem/temp.html')
+# TODO: Prevent joining a full team
+def join_team(request, uid):
+    # Create a new account (without a previously existing
+    # raffle entry) and join an already existing team.
+    team = get_object_or_404(RiverRaftingTeam, join_id=uid)
+    team_leader_name = team.leader.name,
+
```
Comment (dakaza98) [inline src/registrationSystem/views.py:250]: It would be good to add a guard so that if you already belong to a team, you can not join the team

**Review by dakaza98 — CHANGES_REQUESTED** [review]

File: src/registrationSystem/forms.py [hunk 3]
```diff
@@ -61,9 +69,9 @@ class Meta:
         }
 
         widgets = {
-            'name': TextInput(attrs={'readonly': 'readonly'}),
-            'person_nr': TextInput(attrs={'readonly': 'readonly'}),
-            'email': EmailInput(attrs={'readonly': 'readonly'}),
+            'name': TextInput(),
+            'person_nr': TextInput(),
```
Comment (lshoravi) [inline src/registrationSystem/forms.py:73]: I believe TextInput wont give validation here- shouldn't it be PersonNumberInput?

Comment (lshoravi) [reply src/registrationSystem/forms.py:73]: https://github.com/UTNkar/registration-system/blob/development/src/registrationSystem/fields.py

File: src/registrationSystem/views.py [hunk 4]
```diff
@@ -32,7 +31,8 @@ def login_user(request):
         login(request, user)
         return redirect(reverse('overview'))
     else:
-        return redirect(settings.LOGIN_URL)
+        # TODO: Send show nice failed to login page
+        return HttpResponse('Unauthorized', status=401)
```
Comment (lshoravi) [inline src/registrationSystem/views.py:35]: Could you clarify this comment?

File: src/registrationSystem/views.py [hunk 5]
```diff
@@ -191,48 +193,89 @@ def activate(request, token):
...
+                name=form.cleaned_data["name"],
+                email=form.cleaned_data["email"],
+                person_nr=form.cleaned_data["person_nr"],
+                phone_nr=form.cleaned_data["phone_nr"],
+                password=form.cleaned_data["password"],
+                is_utn_member=is_utn_member(form.cleaned_data["person_nr"]),
+            ).save()
+
+            # Create a new team and assign it to the user
+            team = RiverRaftingTeam.objects.create(leader=leader)
```
Comment (lshoravi) [inline src/registrationSystem/views.py:217]: TODO: Use team model depending on environment variables

**Review by lshoravi — CHANGES_REQUESTED** [review]
Small comments- see below

**Pull request merged**